lib/puppet/type/rds_instance.rb

```diff
@@ -0,0 +1,149 @@
+Puppet::Type.newtype(:rds_instance) do
+  @doc = 'type representing an RDS instance'
+
+  ensurable do
+    defaultvalues
+    aliasvalue(:available, :present)
+    newvalue(:destroy) do
+      provider.destroy
+    end
+  end
+
+  newparam(:db_name, namevar: true) do
+    desc 'the name of the db instance'
```

Comment:
So this is deceptive: the db_name is not exactly the namevar: the db_name is the name of the database inside the instance:

```
DBName -> (string)
The meaning of this parameter differs according to the database engine you use. For example, this value returns either MySQL or PostgreSQL information when returning values from create-db-instance-read-replica since Read Replicas are only supported for MySQL and PostgreSQL.
MySQL, SQL Server, PostgreSQL
Contains the name of the initial database of this instance that was provided at create time, if one was specified when the DB instance was created. This same name is returned for the life of the DB instance.
Type: kms-key-id
Oracle
Contains the Oracle System ID (SID) of the created DB instance. Not shown when the returned parameters do not apply to an Oracle DB instance.
```

The namevar is actually `db_instance_identifier`